addons/website_payment/static/src/js/payment_form.js

```diff
@@ -119,4 +123,35 @@ PaymentForm.include({
         } : transactionRouteParams;
     },
 
+    /**
+     * Checks constraints on submit:
+     * 1. The value must be greater than the minimum value.
+     * 2. A radio button must be checked, if the custom amount is selected.
+     * 3. The custom input must have a value.
+     *
+     * @override method from payment.payment_form
+     * @private
+     * @param {Event} ev
+     */
+    async _submitForm(ev) {
```

Comment:
This method order seems random. `_submitForm` should rather go after `_updateAmount`.